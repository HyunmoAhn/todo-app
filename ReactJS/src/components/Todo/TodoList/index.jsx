import React from 'react';
// import PropTypes from 'prop-types';

const propTypes = {};
const defaultProps = {};

class TodoHeader extends React.Component {
  render() {
    return (
      <section className="main">
        <ul className="todo-list">
          <li>
            <div className="view">
              <input type="checkbox" className="toggle" />
              <label>value</label>
              <button className="destroy" />
            </div>
          </li>
          <li className="completed">
            <div className="view">
              <input type="checkbox" className="toggle" />
              <label>value2</label>
              <button className="destroy" />
            </div>
          </li>
          <li className="editing">
            <div className="view">
              <input type="checkbox" className="toggle" />
              <label>value2</label>
              <button className="destroy" />
            </div>
            <input type="text" className="edit" />
          </li>
        </ul>
      </section>
    );
  }
}

TodoHeader.propTypes = propTypes;
TodoHeader.defaultProps = defaultProps;

export default TodoHeader;
