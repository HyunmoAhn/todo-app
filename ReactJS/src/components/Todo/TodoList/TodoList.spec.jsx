import React from 'react';
import { shallow } from 'enzyme';

import TodoList from './';

describe('<TodoList />', () => {
  it('should match to snapshot when render default', () => {
    const todoList = [
      {
        status: 'mock-status',
        value: 'mock-value',
        id: 'mock-id',
      },
    ];
    const wrapper = shallow(<TodoList todoList={todoList} />);

    expect(wrapper).toMatchSnapshot();
  });
});
