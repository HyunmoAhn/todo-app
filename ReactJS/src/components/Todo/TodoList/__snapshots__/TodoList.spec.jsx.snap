// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should match to snapshot when render default 1`] = `
<section
  className="main"
>
  <ul
    className="todo-list"
  >
    <TodoListItem
      item={
        Object {
          "id": "mock-id",
          "status": "mock-status",
          "value": "mock-value",
        }
      }
      key="mock-id"
      onDeleteListItem={[Function]}
      onEditListItemValue={[Function]}
      onEditToggleListItem={[Function]}
      onToggleListItem={[Function]}
    />
  </ul>
</section>
`;
